fdbserver/workloads/ApiWorkload.actor.cpp

```diff
@@ -249,6 +260,18 @@ Key ApiWorkload::generateKey(VectorRef<KeyValueRef> const& data,
 			keyBuffer[0] = deterministicRandom()->randomInt(0, 255);
 	}
 
+	// If encryption validation is enabled; slip "marker pattern" at random location in generated key
+	if (g_simulator->dataAtRestPlaintextMarker.present() &&
```

Comment:
You have to check `g_network->isSimulated()` here too at the start of the `if`.

Reply:
Good catch, thanks for pointing out.